### DISCUSSION TITLE
Getting cookie in the same route where we added it

### DISCUSSION THREAD

**Discussion by szabgab** [rwf2/Rocket#2719] [Questions]
When trying to `get` a cookie in the same route where we `add` it, we get back `None`. Is this the designed behavior?

```rust
#[get("/set")]
fn set_cookie(cookies: &CookieJar<'_>) -> content::RawHtml<String> {
    let data = String::from("data");
    cookies.add(("cookie-demo", data));
    rocket::info!("Cookie after add: {}", cookies.get("cookie-demo").unwrap());
   content::RawHtml("cookie was set")
}
```

**Comment by SergioBenitez** [comment]
Yes. Please see the documentation on pending cookies.

**Reply by szabgab** [reply]
Thanks. For future readers, use [get_pending](https://api.rocket.rs/v0.5/rocket/http/struct.CookieJar.html#method.get_pending) instead of `get` where necessary.
